packages/expo-web-browser/android/src/main/java/expo/modules/webbrowser/WebBrowserModule.kt

```diff
@@ -0,0 +1,212 @@
+package expo.modules.webbrowser
+
+import android.content.Context
+import expo.modules.core.ExportedModule
+import expo.modules.core.interfaces.ExpoMethod
+import expo.modules.webbrowser.error.NoPreferredPackageFound
+import expo.modules.core.errors.CurrentActivityNotFoundException
+import expo.modules.webbrowser.error.PackageManagerNotFoundException
+import expo.modules.core.arguments.ReadableArguments
+import android.content.Intent
+import android.graphics.Color
+import android.net.Uri
+import android.os.Bundle
+import android.text.TextUtils
+import androidx.browser.customtabs.CustomTabsIntent
+import expo.modules.core.ModuleRegistry
+import expo.modules.core.Promise
+import expo.modules.core.utilities.ifNull
+import java.lang.IllegalArgumentException
+
+private const val BROWSER_PACKAGE_KEY = "browserPackage"
+private const val SERVICE_PACKAGE_KEY = "servicePackage"
+private const val BROWSER_PACKAGES_KEY = "browserPackages"
+private const val SERVICE_PACKAGES_KEY = "servicePackages"
+private const val PREFERRED_BROWSER_PACKAGE = "preferredBrowserPackage"
+private const val DEFAULT_BROWSER_PACKAGE = "defaultBrowserPackage"
+private const val SHOW_IN_RECENTS = "showInRecents"
+private const val CREATE_TASK = "createTask"
+private const val DEFAULT_SHARE_MENU_ITEM = "enableDefaultShareMenuItem"
+private const val TOOLBAR_COLOR_KEY = "toolbarColor"
+private const val SECONDARY_TOOLBAR_COLOR_KEY = "secondaryToolbarColor"
+private const val ERROR_CODE = "EXWebBrowser"
+private const val TAG = "ExpoWebBrowser"
+private const val SHOW_TITLE_KEY = "showTitle"
+private const val ENABLE_BAR_COLLAPSING_KEY = "enableBarCollapsing"
+private const val NO_PREFERRED_PACKAGE_MSG = "Cannot determine preferred package without satisfying it."
+
+class WebBrowserModule(context: Context?) : ExportedModule(context) {
+  private lateinit var customTabsResolver: CustomTabsActivitiesHelper
+  private lateinit var connectionHelper: CustomTabsConnectionHelper
+
+  override fun getName() = TAG
+
+  override fun onCreate(moduleRegistry: ModuleRegistry) {
+    customTabsResolver = moduleRegistry.getModule(CustomTabsActivitiesHelper::class.java)
+    connectionHelper = moduleRegistry.getModule(CustomTabsConnectionHelper::class.java)
+  }
+
+  @ExpoMethod
+  fun warmUpAsync(packageName: String?, promise: Promise) {
+    try {
+      val resolvedPackageName = givenOrPreferredPackageName(packageName)
+      connectionHelper.warmUp(resolvedPackageName)
+      val result = Bundle().apply {
+        putString(SERVICE_PACKAGE_KEY, resolvedPackageName)
+      }
```

Comment:
Take advantage of `bundleOf` from https://developer.android.com/kotlin/ktx/extensions-list#top-level_functions_5
```suggestion
      val result = bundleOf(
        SERVICE_PACKAGE_KEY to resolvedPackageName
      )
```